test/functional/options/defaults_spec.lua

```diff
@@ -328,3 +329,64 @@ describe('XDG-based defaults', function()
     end)
   end)
 end)
+
+
+describe('stdpath()', function()
+  before_each(function()
+    clear({env={
+      XDG_CONFIG_HOME='cat',
```

Comment:
Not actually, edge cases are more likely to contain bugs. Though better test *both* realistic and not, I would also suggest using `$XDG_DATA_HOME` and like here (i.e. check that environment variables are not expanded) and `~/foo` (i.e. check that `~/` is not expanded).